src/power.jl

```diff
@@ -195,211 +282,96 @@ function sqrt(a::Taylor1)
         to an **even power** in order to expand `sqrt` around 0."""))
     end
 
-    # Reaching this point, it is possible to implement the sqrt of the Taylor1 polynomial.
     # The last l0nz coefficients are set to zero.
-    ##l0nz > 0 && warn("The last k=$(l0nz) Taylor1 coefficients ARE SET to 0.")
     lnull = div(l0nz, 2)
-    @inbounds aux = sqrt(a[l0nz+1])
+    @inbounds aux = sqrt( a[l0nz+1] )
     T = typeof(aux)
-    v = convert(Array{T,1}, a.coeffs)
-    coeffs = zeros(T, a.order+1)
-    @inbounds coeffs[lnull+1] = aux
+
+    c = Taylor1( zeros(T, a.order+1) )
+    @inbounds c[lnull+1] = aux
     @inbounds for k = lnull+1:a.order-l0nz
-        coeffs[k+1] = sqrtHomogCoef(k, v, coeffs, lnull)
+        sqrt!(c, a, k, lnull)
+    end
+
+    return c
+end
+
+function sqrt(a::TaylorN)
```

Comment:
The catch here is that it is simple to factorize `Taylor1` polynomials of the form `a2 t^2 + a3 t^3 +...` into something like `t^2 (a2 + a3 t +...)` so then taking the `sqrt` makes it simpler. The same trick doesn't apply in a straight forward way to `TaylorN` polynomials. (Yet, `Taylor1{Taylor1{T}}` behaves nicely again!)